### PR TITLE
completed assessment

### PR DESCRIPTION
**Pull request by crgjr66**
somewhat confused on the testing, but forced the test to fail via both the test and code within the function

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -11,26 +11,56 @@ var temp1 = 35
...
+    return `${number} is above boiling point`
+  } else {
+    // is number at boiling point
+    return `${number} is at boiling point`
+  }
+}
+
+console.log(boilingPoint(temp1));
+console.log(boilingPoint(temp2));
+console.log(boilingPoint(temp3));
```
Comment (andeeisaacs) [inline code-challenges.js:30]: Fantastic!

File: code-challenges.js [hunk 2]
```diff
@@ -11,26 +11,56 @@ var temp1 = 35
...
 
 var myNumbers1 = [3, 7, 0, 6, -9]
 
+// multiply each element of an input array
+const multiplyBy5 = (array) => {
+  let newArray = [];
+  // loop through the array
+  for (let i = 0; i < 5; i++) {
+    // multiply each element by 5 and add to new array
+    newArray[i] = array[i] * 5;
```
Comment (andeeisaacs) [inline code-challenges.js:44]: Very interesting way of solving this! Most people would .push new value * 5 into the old array, but I love this solution! Very clean!

File: code-challenges.js [hunk 3]
```diff
@@ -11,26 +11,56 @@ var temp1 = 35
...
+  }
+  return newArray;
+}
 
-
+console.log(multiplyBy5(myNumbers1));
```
Comment (andeeisaacs) [inline code-challenges.js:49]: 👍

File: code-challenges.js [hunk 4]
```diff
@@ -11,26 +11,56 @@ var temp1 = 35
...
 
 // --------------------3) Create a function that multiplies each number in the array by 5 using map.
 // Use the test variable provided below. Expected outcome: [40, -35, 0, 30, 10]
 
 var myNumbers2 = [8, -7, 0, 6, 2]
 
+// function that takes an array parameter and multiplies each element by 5
+const multBy5 = (array) => {
+  // go thru entire array and mult each element by 5
+  return array.map(value => value * 5)
```
Comment (andeeisaacs) [inline code-challenges.js:60]: Nice and clean single return solution!

File: code-challenges.js [hunk 5]
```diff
@@ -39,8 +69,18 @@ var myNumbers2 = [8, -7, 0, 6, 2]
...
+  let strArray = string.split('');
+  // find the vowels and remove them
+  return strArray.filter( value => {
+    return value.toLowerCase() !== 'a' && value.toLowerCase() !== 'e' && value.toLowerCase() !== 'i' && value.toLowerCase() !== 'o' && value.toLowerCase() !== 'u';
+  }).join(''); // reassemble the string before passing it back
+}
 
-
+console.log(removeVowels(stringWithVowels1));
+console.log(removeVowels(stringWithVowels2));
```
Comment (andeeisaacs) [inline code-challenges.js:83]: Beautiful! Love the use of .toLowerCase! To make your code even more concise, you can delcare value = value.toLowerCase() ahead of time before checking for values. 

Example of this:

```
return strArray.filter( value => {
value = value.toLowerCase()
return value !== 'a' && value !== 'e' ... 
``` 

etc! 

Just another way we can keep our code a little more DRY. Great work!

File: code-challenges.js [hunk 6]
```diff
@@ -49,18 +89,59 @@ var stringWithVowels2 = "ILoveJavaScript"
...
+    }).join(''); // reassemble the string before passing it back
+  } else {
+    // if the input was not a string
+    return `${string} is not a string`
+  }
 
+}
 
+console.log(removeVowels2(notAString1));
+console.log(removeVowels2(notAString2));
```
Comment (andeeisaacs) [inline code-challenges.js:111]: Fantastic!

File: code-challenges.js [hunk 7]
```diff
@@ -49,18 +89,59 @@ var stringWithVowels2 = "ILoveJavaScript"
...
+console.log(findTheCats(toonimals));
 
 // --------------------7) Using the toonimals variable from #6, create a function that returns only the names of the non-cats.
 // Expected output: "Itchy" "Daffy" "Ren"
+
+// function to take an array input and return the "cat" elements
+const findTheNonCats = (array) => {
+  // go thru the array and find the NON-cats only
+  return array.filter(value => {
+    return value.animal === "mouse" || value.animal === "duck" || value.animal === "dog";
```
Comment (andeeisaacs) [inline code-challenges.js:136]: This definitely works, for the given object set, but what if there were more than just Mouse, Duck and Dog? We could look at using the bang operator to check for everything NOT cat. Then our function would be much more dynamic for future use.

File: interview-questions.md [hunk 8]
```diff
@@ -7,64 +7,68 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a function? Why would you use one?
 
-  Your answer:
+  Your answer: A function is a javascript method that takes any number of inputs and may/ or may not return some value back to the caller.
```
Comment (andeeisaacs) [inline interview-questions.md:10]: Totally, I'd also mention how a function is reusable and can be dynamic to help cover as to 'WHY' we would use one.

File: interview-questions.md [hunk 9]
```diff
@@ -7,64 +7,68 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer:A JavaScript function is a block of code designed to perform a particular task
 
 
 
 2. What is the difference between .map() and .filter()?
 
-  Your answer:
+  Your answer: .map traverses through an array and returns and array of the same size, .filter goes through each element of an array and returns a subset of the original array.
```
Comment (andeeisaacs) [inline interview-questions.md:18]: 👍

File: interview-questions.md [hunk 10]
```diff
@@ -7,64 +7,68 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: . filter() returns a subset of the elements from the original array, while . map() produces an array with new, different entries based on the elements in the original array.
 
 
 
 3. What is the difference between console.log() and return?
 
-  Your answer:
+  Your answer: console.log() outputs/logs some data to the computer output terminal, where as "return" just returns a value from a function to a calling piece of code.
```
Comment (andeeisaacs) [inline interview-questions.md:26]: Absolutely. Great answer.

File: interview-questions.md [hunk 11]
```diff
@@ -7,64 +7,68 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: Return is used to interrupt the flow of execution of a function and return some value (or not). console. log is used to print information to the console
 
 
 
 4. In regards to functions, what is an argument?
 
-  Your answer:
+  Your answer: An argument is a values that is given to the function in the parameter list so that the function can use the data to either manipulate, modify, or calculate some responding data.
```
Comment (andeeisaacs) [inline interview-questions.md:34]: 100%, wonderful answer.

File: interview-questions.md [hunk 12]
```diff
@@ -7,64 +7,68 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: The parameters, in a function call, are the function's arguments. JavaScript arguments are passed by value: The function only gets to know the values, not the argument's locations. If a function changes an argument's value, it does not change the parameter's original value.
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: The "driver" has the responsibility of writing the code, the "navigator" is responsible for defining the requirements and reviewing the code as it is being written.
```
Comment (andeeisaacs) [inline interview-questions.md:42]: Totally nailed it with the definitions of the roles. I'd maybe mention a little more as to its purpose and techniques. I'd maybe also mention how the roles can be swapped over periods of time.

File: interview-questions.md [hunk 13]
```diff
@@ -7,64 +7,68 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: Pair programming is an agile software development technique in which two programmers work together at one workstation. One, the driver, writes code while the other, the observer or navigator, reviews each line of code as it is typed in. The two programmers switch roles frequently.
 
 
 
 6. What is TDD? Describe the work flow associated with TDD.
 
-  Your answer:
+  Your answer: "TDD" is Test Driven Development.  This is where the developer writes the tests before any actual functional code is developed.  Work flow = Write test, write code, run test, modify, success.
```
Comment (andeeisaacs) [inline interview-questions.md:50]: Yes! 🙌  The name for the flow is called 'Red-Green-Refactoring'

File: testing-with-jest.test.js [hunk 14]
```diff
@@ -6,22 +6,66 @@
...
+    expect(addThemUp(2, 3)).toEqual(5)
+    expect(addThemUp(2, 10)).toEqual(12)
+    expect(addThemUp(-2, 2)).toEqual(0)
+  })
+})
 
+const addThemUp = (num1, num2) => {
+  //return num1 - num2;
+  return num1 + num2;
+}
```
Comment (andeeisaacs) [inline testing-with-jest.test.js:22]: Excellent!

File: testing-with-jest.test.js [hunk 15]
```diff
@@ -6,22 +6,66 @@
...
+describe("triangleArea", () => {
+  test("take base and height of triangle and return the area", () => {
+    expect(triangleArea(2, 3)).toEqual(3)
+    expect(triangleArea(2, 10)).toEqual(10)
+  })
+})
 
+const triangleArea = (base, height) => {
+  return base * height / 2;
+}
```
Comment (andeeisaacs) [inline testing-with-jest.test.js:55]: Great!

**Review by andeeisaacs — APPROVED** [review]
Chuck, your assessment is fantastic! I'd be really proud of this work if I were you. 

- Your pseudo coding was clear and easy to follow. I would encourage you to keep it up! 
- Your tests were great! I think you just got a bit confused between what to do with the 1a-1b. In the 1 a section, we write the test, in the 1-b section, we write the code that makes the test pass. But its totally fine for right now. If you ever have questions about this kind of stuff before submitting, feel free to reach out to us ahead of time so we can clear it up! 
- I would encourage you, when creating functions, to think about how to keep them dynamic and reusable not only for the data you are currently using, but for the unknowns of different data.  An example of this is the refactored version of checking the animal type challenge. 

Excellent work, feel free to merge to master!

**Comment by crgjr66** [comment]
reviewed comments and knew I forgot array.length after I submitted, rather than the exact number of elements in the array, arghh

**Pull request merged**